### PR TITLE
Use of ECMWF versions of ATLAS and FCKIT

### PR DESCRIPTION
**Pull request by benjaminmenetrier**
This PR includes corrections to use ECMWF versions of ATLAS and FCKIT, see https://github.com/orgs/JCSDA/teams/jedi-core/discussions/11.

Test changes are due to a different Earth radius used in `src/soca/Transforms/HorizFilt/soca_horizfilt_mod.F90`: [6371229 m](https://github.com/ecmwf/atlas/blob/37faae7df2f565ad48293e635cd492d3d4e787d8/src/atlas/util/Earth.h#L24) instead of 6371000 m.

### CODE REVIEW

File: src/soca/Transforms/HorizFilt/soca_horizfilt_mod.F90 [hunk 1]
```diff
@@ -5,6 +5,7 @@
 
 
 module soca_horizfilt_mod
+  use atlas_module
```
Comment (guillaumevernieres) [inline src/soca/Transforms/HorizFilt/soca_horizfilt_mod.F90]: Add [...], only: atlas_geometry, atlas_indexkdtree, ...

File: src/soca/Transforms/HorizFilt/soca_horizfilt_mod.F90 [hunk 2]
```diff
@@ -83,6 +83,9 @@ subroutine soca_horizfilt_setup(self, f_conf, geom, traj, vars)
     ! Allocate and compute filtering weights
     allocate(self%wgh(self%isd:self%ied,self%jsd:self%jed,-1:1,-1:1))
 
+    ! Create UnitSphere geometry
+    ageometry = atlas_geometry("Earth")
```
Comment (travissluka) [inline src/soca/Transforms/HorizFilt/soca_horizfilt_mod.F90:86]: oooh, can I put in "Mars" for Martian planet radius?? (serious question)

Comment (benjaminmenetrier) [reply src/soca/Transforms/HorizFilt/soca_horizfilt_mod.F90:86]: Actually you can put "UnitSphere", "Earth" or any positive real value (including the Martian planet radius). I used it in the tests for instance: https://github.com/ecmwf/atlas/blob/5b0f56cb3d7f5314e4143f8cbcf99d280d39e897/src/tests/util/fctest_geometry.F90#L130.

We could add an alias "Mars"... I'm not sure ECMWF will agree but it's technically feasible!

**Comment by guillaumevernieres** [comment]
Thanks for being on top of this @benjaminmenetrier , much appreciated!

**Comment by guillaumevernieres** [comment]
@benjaminmenetrier : I removed the branch matching for atlas ... so don't worry too much about the travis-ci failure for now.

**Review by guillaumevernieres — APPROVED** [review]

**Pull request merged**